solidity/contracts/deposit/DepositLiquidation.sol

```diff
@@ -61,42 +61,37 @@ library DepositLiquidation {
         return (_bondValue.mul(100).div(_lotValue));
     }
 
-    /// @notice         Starts signer liquidation due to fraud.
-    /// @dev            Liquidation is done by auction.
-    /// @param  _d      Deposit storage pointer.
-    function startSignerFraudLiquidation(DepositUtils.Deposit storage _d) internal {
-        _d.logStartedLiquidation(true);
+    /// @notice           Starts signer liquidation due to abort, undercollateralization, or fraud
+    /// @dev              Liquidation is done by auction.
+    /// @param _wasFraud  Boolean checking liquidation cause. True if fraud, false otherwise.
+    /// @param _d         Deposit storage pointer.
+    function startLiquidation(DepositUtils.Deposit storage _d, bool _wasFraud) internal {
+        _d.logStartedLiquidation(_wasFraud);
 
-        // Reclaim used state for gas savings
-        _d.redemptionTeardown();
         uint256 _seized = _d.seizeSignerBonds();
 
-        if (_d.auctionTBTCAmount() == 0) {
-            // we came from the redemption flow
+        // if we come from the redemption flow we shouldn't go to auction.
+        // Instead give the signer bonds to redeemer
+        if (_d.inRedemption()) {
             _d.setLiquidated();
             _d.enableWithdrawal(_d.redeemerAddress, _seized);
             _d.logLiquidated();
+            // Reclaim used state for gas savings
+            _d.redemptionTeardown();
```

Comment:
We can still do this once alongside the bond seizure, right? `redemptionTeardown` doesn't change the deposit state, so the `_d.inRedemption()` check should still work.

Reply:
`redemptionTeardown()` zeroes out `redeemerAddress`
This is new, I just added it to claim more state back.

Reply:
I just saw comment below...